### PR TITLE
gh-107954: Add PyConfig_MEMBER_BOOL type to PyConfigSpec

### PR DESCRIPTION
**Pull request by vstinner**
_PyConfig_AsDict() now returns bool objects for options using the new PyConfig_MEMBER_BOOL type.

Update tests for these changes.

* Issue: gh-107954

### CODE REVIEW

**Comment by vstinner** [comment]
cc @swtaarrs

File: Lib/test/test_embed.py [hunk 1]
```diff
@@ -443,31 +443,31 @@ class InitConfigTests(EmbeddingTestsMixin, unittest.TestCase):
...
         'int_max_str_digits': sys.int_info.default_max_str_digits,
-        'cpu_count': -1,
-        'faulthandler': 0,
-        'tracemalloc': 0,
-        'perf_profiling': 0,
-        'import_time': 0,
-        'code_debug_ranges': 1,
-        'show_ref_count': 0,
-        'dump_refs': 0,
+        'cpu_count': -True,
```
Comment (swtaarrs) [inline Lib/test/test_embed.py]: `cpu_count` is still an int, so I think this change is wrong.

File: Python/initconfig.c [hunk 2]
```diff
@@ -1285,19 +1294,20 @@ _PyConfig_FromDict(PyConfig *config, PyObject *dict)
...
-                config_dict_invalid_value(spec->name);
-                return -1;
+            if (spec->type == PyConfig_MEMBER_BOOL
+                || spec->type == PyConfig_MEMBER_UINT)
+            {
+                if (value < 0) {
+                    config_dict_invalid_value(spec->name);
+                    return -1;
+                }
             }
```
Comment (swtaarrs) [inline Python/initconfig.c:1311]: Would it be worth also verifying that `value <= 1` for `BOOL`?

Comment (vstinner) [reply Python/initconfig.c:1311]: Currently, PyConfig_Read() doesn't normalize boolean values to (0, 1). I prefer to tolerate values larger than 1 for now.

The main reason for this change is to return bool in _PyConfig_AsDict().

File: Lib/test/test_embed.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/test_embed.py]: Oops, fixed. I blindly replaced 0 with False and 1 with True :-)

**Pull request merged**